Mathlib/RingTheory/HahnSeries/Binomial.lean

```diff
@@ -0,0 +1,170 @@
+        hx r b hg) fun b hb => coeff_eq_zero_of_lt_orderTop <| binomialFamily_orderTop_pos hx r <|
+        Nat.zero_lt_of_ne_zero hb
+    · have : (binomialFamily x r 0).coeff 0 = 1 := by simp [hx]
+      rw [← this]
+      refine hsum_leadingCoeff_of_le (g := 0) (a := 0) (by simp [hx]) ?_ ?_
+      · intro b g' hg'
+        exact binomialFamily_mem_support hx r b hg'
+      · intro b hb
+        exact coeff_eq_zero_of_lt_orderTop <| binomialFamily_orderTop_pos hx r <|
+        Nat.zero_lt_of_ne_zero hb
+
+end SummableFamily
+
+/-- The group of invertible Hahn series close to 1, i.e., those series such that subtracting 1
+  yields a series with strictly positive `orderTop`. -/
+@[simps]
+def orderTopSubOnePos (Γ) (R) [LinearOrder Γ] [AddCommMonoid Γ] [IsOrderedCancelAddMonoid Γ]
+    [CommRing R] :
+    Subgroup (HahnSeries Γ R)ˣ where
+  carrier := { x : (HahnSeries Γ R)ˣ | 0 < (x.val - 1).orderTop}
+  mul_mem' := by
+    intro x y hx hy
+    obtain (_|_) := subsingleton_or_nontrivial R
+    · simp
+    · simp_all only [Set.mem_setOf_eq, orderTop_self_sub_one_pos_iff]
+      have h1 : x.val.leadingCoeff * y.val.leadingCoeff = 1 := by rw [hx.2, hy.2, mul_one]
+      constructor
+      · rw [Units.val_mul, orderTop_mul_of_nonzero (by simp [h1]), hx.1, hy.1, add_zero]
+      · rw [Units.val_mul, leadingCoeff_mul_of_nonzero (h1 ▸ one_ne_zero), h1]
+  one_mem' := by simp
+  inv_mem' {y} h := by
+    suffices 0 < (y.inv - 1).orderTop by exact this
+    obtain (_|_) := subsingleton_or_nontrivial R
+    · simp
+    · have : 0 < (y.val - 1).orderTop := h
+      rw [orderTop_self_sub_one_pos_iff] at this
+      have nz : y.val.leadingCoeff * y.inv.leadingCoeff ≠ 0 := by
+        rw [this.2, one_mul]
+        exact leadingCoeff_ne_zero.mpr (by simp)
+      refine y.inv.orderTop_self_sub_one_pos_iff.mpr ⟨?_, ?_⟩
+      · simpa [this.1, y.val_inv] using (orderTop_mul_of_nonzero nz).symm
+      · simpa [this.2, y.val_inv] using (leadingCoeff_mul_of_nonzero nz).symm
+
+@[simp]
+theorem mem_orderTopSubOnePos_iff (x : (HahnSeries Γ R)ˣ) :
+    x ∈ orderTopSubOnePos Γ R ↔ 0 < (x.val - 1).orderTop := by
+  exact Eq.to_iff rfl
+
+/-- Make an element of `orderTopSubOnePos` -/
+@[simps]
+def toOrderTopSubOnePos {x : HahnSeries Γ R} (h : 0 < (x - 1).orderTop) :
+    orderTopSubOnePos Γ R where
+  val := ⟨x, (isUnit_of_orderTop_pos h).unit.inv, IsUnit.mul_val_inv (isUnit_of_orderTop_pos h),
+    IsUnit.val_inv_mul (isUnit_of_orderTop_pos h)⟩
+  property := h
+
+theorem one_plus_single_mem_orderTopSubOnePos {g : Γ} (hg : 0 < g) (r : R) :
+    (toOrderTopSubOnePos (orderTop_sub_pos hg r)).val ∈ orderTopSubOnePos Γ R :=
+  (mem_orderTopSubOnePos_iff (toOrderTopSubOnePos (orderTop_sub_pos hg r)).val).mpr
+    (orderTop_sub_pos hg r)
+
+open SummableFamily
+
+instance [BinomialRing R] :
+    HPow (orderTopSubOnePos Γ R) R (orderTopSubOnePos Γ R) where
+  hPow x r := toOrderTopSubOnePos (orderTop_hsum_binomialFamily_pos x.2 r)
```

Comment:
```suggestion
instance [BinomialRing R] :
    Pow (orderTopSubOnePos Γ R) R where
  pow x r := toOrderTopSubOnePos (orderTop_hsum_binomialFamily_pos x.2 r)
```
Unless there a specific reason why you are avoiding `Pow`. If so it should be explained in a docstring.